docs/website/docs/plus/getting-started/advanced_tutorial.md

```diff
@@ -0,0 +1,305 @@
+
+
+```yaml
+# your sources are the data sources you want to load from
+sources:
+  arrow:
+    type: sources.arrow.source
+
+# your destinations are the databases where your data will be saved
+destinations:
+  duckdb:
+    type: duckdb
+
+datasets:
+  my_pipeline_dataset:
+    destination:
+        - bigquery
+
+# your pipelines orchestrate data loading actions
+pipelines:
+  my_pipeline:
+    source: arrow
+    destination: duckdb
+    dataset_name: my_pipeline_dataset
+```
+
+If you try to run the pipeline "my_pipeline" now, you will receive an error that the dataset `my_pipeline_dataset` does not exist on the bigquery destination, you may allow this by adding `bigquery` to the allowed destinations of `my_pipeline_dataset`.
+
+```yaml
+my_pipeline_dataset:
+  destination:
+      - duckdb
+      - bigquery
+```
+
+:::tip
+If  you run dataset cli commands without providing a destination name, dlt+ will always select the first destination in the list by default. If you allow implicit entities, dlt+ can also discover datasets only defined on pipelines and will use the destination of that pipeline for instantiating the dataset. The same is true when retrieving datasets from the catalog in python, ore on that below.
+:::
+
+
+## Interacting with the project context in code
+
+You can also interact with your dlt+ project directly in code as opposed to running cli commands. Let's use the same starting point as above by running `dlt project init arrow duckdb`. Now add a new python file in the root of your project called `do_something.py` and add the following code:
+
+```py
+from dlt_plus import current
+
+if __name__ == "__main__":
+    # this will get the currently active project and print its name
+    print(current.project().name)
+```
+
+Run the above with `python do_something.py` and see the output.
+
+The current module will allow you to interact with the currently active project. You can retrieve the project configuration with `current.project()`, get an instantiatend version of all defined entities from the entities factory at `current.entities()`, access all defined datasets in the catalog: `current.catalog()`, and run pipelines with the runner at `current.runner()`.
+
+### Accessing project settings
+
+A few examples of what you have access too on the project object:
+
+```py
+# show the currently active profile
+print(current.project().current_profile)
+# show the main project dir
+print(current.project().project_dir)
+# show the project config
+print(current.project().config)
+# list explicitely defined datasets (also works with destinations, sources, pipelines etc.)
+print(current.project().datasets)
+```
+
+### Accessing entities
+
+Accessing entities works in the same way as when accessing entities in the `dlt.yml`. If possible and allowed, implicit entities are created and returned back to you, if you, an error will be raised.
+
+```py
+# get a pipeline instance
+pipeline = current.entities().create_pipeline("my_pipeline")
```

Comment:
it took me a bit of back and forth until i got that this is only creating the pipeline instance (or entity) and not a new one